cpp/Platform.Collections/Stacks/CStack.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace Platform::Collections::Stacks
+{
+    template<typename TSelf,typename TElement>
+    concept CStack =
+        requires(TSelf& self,TElement item) 
+        {
+            { self.empty() } -> std::same_as<bool>;
+
+            { self.push(item) };
+
+            { self.pop() };
+
+            { self.top() } -> std::same_as<TElement&>;
```

Comment:
Same question here.

Reply:
Not same. `TElement` reference is mutable